File: examples/php-net/Tutorial02.php
<?php
	/*==================================================================
	 | Tutorial 02 
	 |
	 | This code sample shows how to export list to Excel file in PHP.
	 | The list contains data from a SQL database.
	 | The cells are formatted using a user-defined format.
	  ================================================================*/
  	
	include("Color.inc");
	include("Alignment.inc");

    // Constants declaration
    $OddRowStripesStyleColor = 0xf0f7ef;
    
	header("Content-Type: text/html");
	
	echo "Tutorial 02<br>";
	echo "----------<br>";
		
	// Create an instance of the class that exports Excel files
	$workbook = new COM("EasyXLS.ExcelDocument");

	// Create the database connection
	$serverName = "(local)";
	$connectionInfo = array("Database"=>"northwind","UID"=>"sa","PWD"=>"");
	
	$db_conn = sqlsrv_connect( $serverName, $connectionInfo); 
	if( $db_conn === false )  
	{
   	  echo "Unable to connect.";
  	   die( print_r( sqlsrv_errors(), true));
	}

	// Query the database
	$query_result = sqlsrv_query( $db_conn, "SELECT TOP 100 CAST(Month(ord.OrderDate) AS varchar)+'/' + CAST(Day(ord.OrderDate) AS varchar) + '/' + CAST(year(ord.OrderDate) AS varchar) AS 'Order Date', P.ProductName AS 'Product Name', O.UnitPrice AS Price, ' ' + cast(O.Quantity AS varchar) AS Quantity , O.UnitPrice * O. Quantity AS Value FROM Orders AS ord, [Order Details] AS O, Products AS P WHERE 	O.ProductID = P.ProductID AND O.OrderID = ord.OrderID" )
		or die( "<strong>ERROR: Query failed</strong>" );

	// Create the list that stores the query values
	$lstRows = new COM("EasyXLS.Util.List");
	
	// Add the report header row to the list
	$lstHeaderRow  = new COM("EasyXLS.Util.List");
	$lstHeaderRow->addElement("Order Date");
	$lstHeaderRow->addElement("Product Name");
	$lstHeaderRow->addElement("Price");
	$lstHeaderRow->addElement("Quantity");
	$lstHeaderRow->addElement("Value");
	$lstRows->addElement($lstHeaderRow);
			
	// Add the query values from the database to the list
	while ($row=sqlsrv_fetch_array($query_result))
	{
		$RowList = new COM("EasyXLS.Util.List");
		$RowList->addElement("" . $row['Order Date']);
		$RowList->addElement("" . $row["Product Name"]);
		$RowList->addElement("" . $row["Price"]);
		$RowList->addElement("" . $row["Quantity"]);
		$RowList->addElement("" . $row["Value"]);
		$lstRows->addElement($RowList);
			
	}
	
	// Create an instance of the class used to format the cells in the report
	$xlsAutoFormat = new COM("EasyXLS.ExcelAutoFormat");
	
	// Set the formatting style of the header
	$xlsHeaderStyle = new COM("EasyXLS.ExcelStyle");
	$xlsHeaderStyle->setBackground((int)$COLOR_LIGHTGREEN);
	$xlsHeaderStyle->setFontSize(12);
	$xlsAutoFormat->setHeaderRowStyle($xlsHeaderStyle);

	// Set the formatting style of the cells (alternating style)
	$xlsEvenRowStripesStyle = new COM("EasyXLS.ExcelStyle");
	$xlsEvenRowStripesStyle->setBackground((int)$COLOR_FLORALWHITE);
	$xlsEvenRowStripesStyle->setFormat("$0.00");
	$xlsEvenRowStripesStyle->setHorizontalAlignment($ALIGNMENT_ALIGNMENT_LEFT);
	$xlsAutoFormat->setEvenRowStripesStyle($xlsEvenRowStripesStyle)	;
	$xlsOddRowStripesStyle = new COM("EasyXLS.ExcelStyle");
	$xlsOddRowStripesStyle->setBackground((int)$OddRowStripesStyleColor);
	$xlsOddRowStripesStyle->setFormat("$0.00");
	$xlsOddRowStripesStyle->setHorizontalAlignment ($ALIGNMENT_ALIGNMENT_LEFT);
	$xlsAutoFormat->setOddRowStripesStyle($xlsOddRowStripesStyle);
	$xlsLeftColumnStyle = new COM("EasyXLS.ExcelStyle");
	$xlsLeftColumnStyle->setBackground((int)$COLOR_FLORALWHITE);
	$xlsLeftColumnStyle->setFormat("mm/dd/yyyy");
	$xlsLeftColumnStyle->setHorizontalAlignment($ALIGNMENT_ALIGNMENT_LEFT);
	$xlsAutoFormat->setLeftColumnStyle($xlsLeftColumnStyle);
	
	// Export list to Excel file
	echo "Writing file: C:\Samples\Tutorial02 - export List to Excel with formatting.xlsx<br>";
	$workbook->easy_WriteXLSXFile_FromList_2("C:\Samples\Tutorial02 - export List to Excel with formatting.xlsx", $lstRows, $xlsAutoFormat, "Sheet1");
	
	// Confirm export of Excel file
	if ($workbook->easy_getError() == "")
		echo "File successfully created.";
	else
		echo "Error encountered: " . $workbook->easy_getError();
		
	// Free the memory associated with the query
	sqlsrv_free_stmt( $query_result );

	// Close database connection
	sqlsrv_close( $db_conn );           
  
  	// Dispose memory
	$workbook->Dispose();

?>

File: examples/python-net/Tutorial02.py
"""--------------------------------------------------------------------
Tutorial 02

This code sample shows how to export List to Excel file in Python.
The List contains data from a SQL database.
The cells are formatted using a user-defined format.
--------------------------------------------------------------------"""

import clr
import gc
import sqlite3

clr.AddReference('EasyXLS')
from EasyXLS import *
from EasyXLS.Util import *
from EasyXLS.Constants import *
from System.Drawing import *

print("Tutorial 02\n-----------\n")

# Create an instance of the class that exports Excel files
workbook = ExcelDocument()

# Create the database connection
conn = sqlite3.connect('chinook.db')

# Query the database
cursor = conn.execute("SELECT strftime('%m/%d/%Y', I.InvoiceDate), C.FirstName || ' ' || C.LastName, I.BillingAddress, I.BillingCity, I.BillingState, I.BillingCountry, I.Total FROM Invoices I INNER JOIN Customers C ON I.CustomerId=C.CustomerId LIMIT 100")
rows = cursor.fetchall()

# Create the list that stores the query values
lstRows = List()

# Add the report header row to the list
lstHeaderRow = List()
lstHeaderRow.addElement("Invoice date")
lstHeaderRow.addElement("Customer name")
lstHeaderRow.addElement("Billing address")
lstHeaderRow.addElement("Billing city")
lstHeaderRow.addElement("Billing state")
lstHeaderRow.addElement("Billing country")	
lstHeaderRow.addElement("Total")	
lstRows.addElement(lstHeaderRow)

# Add the query values from the database to the list
for row in rows:
    RowList = List()
    RowList.addElement(row[0])
    RowList.addElement(row[1])	
    RowList.addElement(row[2])
    RowList.addElement(row[3])
    RowList.addElement(row[4])
    RowList.addElement(row[5])
    RowList.addElement(row[6])
    lstRows.addElement(RowList)

# Create an instance of the class used to format the cells in the report
xlsAutoFormat = ExcelAutoFormat()
        
# Set the formatting style of the header
xlsHeaderStyle = ExcelStyle(Color.LightGreen)
xlsHeaderStyle.setFontSize(12)
xlsAutoFormat.setHeaderRowStyle(xlsHeaderStyle)

# Set the formatting style of the cells (alternating style)
xlsEvenRowStripesStyle = ExcelStyle(Color.FloralWhite)
xlsEvenRowStripesStyle.setFormat("$0.00")
xlsEvenRowStripesStyle.setHorizontalAlignment(Alignment.ALIGNMENT_LEFT)
xlsAutoFormat.setEvenRowStripesStyle(xlsEvenRowStripesStyle)
xlsOddRowStripesStyle = ExcelStyle(Color.FromArgb(240, 247, 239))
xlsOddRowStripesStyle.setFormat("$0.00")
xlsOddRowStripesStyle.setHorizontalAlignment (Alignment.ALIGNMENT_LEFT)
xlsAutoFormat.setOddRowStripesStyle(xlsOddRowStripesStyle)
xlsLeftColumnStyle = ExcelStyle(Color.FloralWhite)
xlsLeftColumnStyle.setFormat("mm/dd/yyyy")
xlsLeftColumnStyle.setHorizontalAlignment (Alignment.ALIGNMENT_LEFT)
xlsAutoFormat.setLeftColumnStyle(xlsLeftColumnStyle)

# Export the Excel file
print("Writing file C:\\Samples\\Tutorial02 - export List to Excel with formatting.xlsx.")
workbook.easy_WriteXLSXFile_FromList("c:\\Samples\\Tutorial02 - export List to Excel with formatting.xlsx", lstRows, xlsAutoFormat, "Sheet1")

# Confirm export of Excel file
sError = workbook.easy_getError()

if sError == "":
    print("\nFile successfully created.\n\n")
else:
    print("\nError encountered: " + sError + "\n\n")

# Close the database connection
cursor.close()
conn.close()

# Dispose memory
gc.collect()